agent/osv_output_handler.py

```diff
@@ -172,9 +168,79 @@ def parse_vulnerabilities(
         return []
 
 
-def _aggregate_cves(cve_ids: list[str], api_key: str | None = None) -> Tuple[str, str]:
+def parse_api_vulnerabilities(
+    output: dict[str, Any],
+    package_name: str,
+    package_version: str,
+    api_key: str | None = None,
+) -> list[VulnData]:
+    """Parse the OSV API response to extract vulnerabilities.
+    Args:
+        output: The API response json.
+        package_name: The package name.
+        package_version: The package version.
+        api_key: The NVD API key.
+    Returns:
+        Parsed output.
+    """
+    cves_list: List[str] = []
+    risks_list: List[str] = []
+    vulnerabilities = output.get("vulns", []) or output.get("vulnerabilities", [])
+    summary = ""
+    fixed_versions: list[str] = []
+    cvss_v3_vector = ""
+    references: List[dict[str, Any]] = []
+    description = ""
+    for vulnerability in vulnerabilities:
+        risk = vulnerability.get("database_specific", {}).get("severity")
+        fixed_version = _get_fixed_version(vulnerability.get("affected"))
+        if fixed_version != "":
+            fixed_versions.append(fixed_version)
+        filtered_cves = [
+            alias for alias in vulnerability.get("aliases", []) if "CVE" in alias
+        ]
+        for cve in filtered_cves:
+            description += f"- [{cve}]({CVE_MITRE_URL}{cve}) "
+            description += f": {vulnerability.get('details')}\n"
+        if risk is None:
+            risk_ratings = [
+                cve_service_api.get_cve_data_from_api(cve, api_key).risk
```

Comment:
the part where we compute the risk using the API should be in a separate function and resued between the two functions